src/wp-includes/comment-template.php

```diff
@@ -1942,7 +1942,14 @@ function get_comment_id_fields( $post_id = 0 ) {
 
 	$reply_to_id = isset( $_GET['replytocom'] ) ? (int) $_GET['replytocom'] : 0;
 	$result      = "<input type='hidden' name='comment_post_ID' value='$post_id' id='comment_post_ID' />\n";
-	$result     .= "<input type='hidden' name='comment_parent' id='comment_parent' value='$reply_to_id' />\n";
+
+	if ( 0 !== $reply_to_id ) {
+		$comment = get_comment( $reply_to_id );
+
+		if ( 0 !== (int) $comment->comment_approved ) {
+			$result .= "<input type='hidden' name='comment_parent' id='comment_parent' value='$reply_to_id' />\n";
+		}
```

Comment:
Along with not validating the `comment_parent` is approved, I've noticed it doesn't verify the comment parent's post matches `$post_id`, so it might be possible to reply to comment X on post Y from post Z.

If validation is to be introduced, I guess it would be good to do that too. Are you able to add that in?

Reply:
Great idea! I've added this now - along with additional tests as suggested. Thanks @peterwilsoncc!